nix/modules/writers.nix

```diff
@@ -0,0 +1,81 @@
+{
```

Comment:
could this live somewhere in a library repo that is maintained by the nix-community? i'm hesitant to take it on as i also don't fully understand it as of now, but it seems to reimplement some of `nix-shell --pure`.